python-lib/wordcloud_visualizer.py

```diff
@@ -0,0 +1,267 @@
+        """Return a wordcloud as a matplotlib figure"""
+        # Manage font exceptions based on language
+        font = self._retrieve_font(language)
+        font_path = os.path.join(self.font_path, font)
+        # Generate wordcloud
+        wc = self._get_wordcloud(frequencies, font_path)
+        fig = plt.figure(figsize=self.figsize, dpi=self.dpi)
+        fig.tight_layout()
+        plt.axis("off")
+        plt.rcParams["axes.titlepad"] = self.titlepad
+        plt.rcParams["axes.titlesize"] = self.titlesize
+        plt.title(title)
+        plt.imshow(wc, interpolation="bilinear")
+        return fig
+
+    @time_logging(log_message="Preparing data")
+    def _prepare_data(self, df: pd.DataFrame) -> list:
+        """Private method to reshape data depending on language and subcharts settings
+        Args:
+            df: dataframe containing a text column and, optionally, language and subcharts columns
+        Returns:
+            List of pandas dataframes with one dataframe per language per subchart
+        """
+        if self.subchart_column or self.language_column:
+            # Group data per language and subchart for tokenization
+            group_columns = [col for col in [self.language_column, self.subchart_column] if col]
+            df.dropna(subset=group_columns, inplace=True)
+            df_grouped = df.groupby(group_columns)
+        else:
+            # Simply format data similarly
+            df_grouped = [(self.language, df)]
+
+        return df_grouped
+
+    def _tokenize_texts(self, df_grouped: list) -> list:
+        """Private method to tokenize each group of observations in its correct language
+        Args:
+            df_grouped: list of pandas dataframes with one dataframe per language per subchart
+        Returns:
+            List of spacy docs
+        """
+        # Get language and subchart name for each group
+        texts = []
+        group_names = []
+        for name, group in df_grouped:
+            texts.append([group[self.text_column].str.cat(sep=" ")])
+            group_names.append(name)
+
+        # Get tokenization languages differently depending on language/subchart settings combinations
+        if not self.language_column and not self.subchart_column:
+            languages = [self.language]
+        elif self.language_column and self.subchart_column:
+            languages, subcharts = zip(*group_names)
+            languages = list(languages)
+            self.subcharts = list(subcharts)
+        elif self.subchart_column:
+            self.subcharts = group_names
+            languages = [self.language] * len(self.subcharts)
+        else:
+            languages = group_names
+
+        # Tokenize
+        docs = [self.tokenizer.tokenize_list(text, language)[0] for text, language in zip(texts, languages)]
+
+        return docs
+
+    @time_logging(log_message="Counting tokens")
+    def _count_tokens(self, docs: list) -> list:
```

Comment:
A better type hint would be
```suggestion
    def _count_tokens(self, docs: List[Doc]) -> List[Tuple[AnyStr, Dict]]:
```
This way it is clear for a maintainer what data structure this function assumes

You may need to import stuff from the `typing` base library, as well as `from spacy.tokens import Doc`